android/lib/src/main/java/org/vosk/android/SpeechService.java

```diff
@@ -258,9 +263,17 @@ public void run() {
 
                     //Checks if the queue is full
                     if (chunksInQueue >= RECOGNIZER_QUEUE) {
-                        listener.onTranscriptionFailed();
+                        transcriptionThread.isToResetRecognizer();
+                        isToStopRecognizing = true;
```

Comment:
I am not quite understanding this flag isToStopRecognizing. We set it to true but don't do anything else, so the queue is cleared but all "whiles" continue and we just signal that when we do indeed finish (which is when we interrupt the thread or call something from outside which can take a while) we want to call onRecognizerStopped()?
Perhaps I am missing something? 🤔

Reply:
The isToStopRecognizing flag works to signal outside of the loop that the recognizer has stopped.  So when we leave the while loop due to an interruption of the thread, we only send the callback onRecognizerStopped() when the recognizer is going to be reset. Probably this can be changed and this protection can be performed outside on the listener.

Reply:
I haven't seen the logic on the DT (the client) so is this what we have?
* receive onTranscriptionFailed() and there's an internal variable saving that we stopped the recording
* call something to stop the thread
* since the variable was set to true once the thread stops we get the onRecognizerStopped() to know that it was stopped? 

If that is the case (just guessing, it may not be) - then we should register that we want to call onRecognizerStopped() when the client asks to stop and not when we signal onTranscriptionFailed()

Reply:
The logic was on this way:
- Receive the onTrasciptionFailed() on the DT side;
- Call stopService on the DT side;
- Receive callback of the onRegognizerStopped;
- Call the start again; 

What I was suggesting is: 
- when we call on the DT side the stop inside the onTranscriptionFailed() have a flag that will indicate on the onRecognizerStopped() that we must restart the service